kernelci/db/kernelci_api.py

```diff
@@ -108,6 +108,20 @@ def get_regressions(self, regression_name):
         resp = self._get('?'.join(['regressions', 'name=' + regression_name]))
         return json.loads(resp.text)
 
+    def get_completed_event(self, node_id, hours=0, minutes=0, seconds=0):
+        """Get an event when all child nodes are complete of a given node"""
+        params = {
+            'wait_time_hours': hours,
+            'wait_time_minutes': minutes,
+            'wait_time_seconds': seconds
+        }
```

Comment:
Sorry I didn't notice this before, and I see the API parts have been merged already.  I think we should remove this completely, we can't have a query that waits on the API side when we should just have a timer on the client side.

Reply:
So what I need here is to move the logic of `Node.wait_for_node` in the pipeline.
The endpoint will just check whether the node is completed or not and trigger event.

Reply:
Setting node status to `timeout` will be handled by `set_timeout` script.
I will update the `trigger_completed` pipeline script to wait for the child nodes to get completed.
When all child nodes are completed, the root status will be set to completed and the API endpoint will be called only to trigger `completed` event at the end.